domain/model/service/modelservice.go

```diff
@@ -50,22 +52,33 @@ type ModelService struct {
 	modelID      coremodel.UUID
 	controllerSt ControllerState
 	modelSt      ModelState
+	provider     providertracker.ProviderGetter[environs.Environ]
 }
 
 // NewModelService returns a new Service for interacting with a models state.
 func NewModelService(
 	modelID coremodel.UUID,
 	controllerSt ControllerState,
 	modelSt ModelState,
+	provider providertracker.ProviderGetter[environs.Environ],
 ) *ModelService {
 	return &ModelService{
 		modelID:      modelID,
 		controllerSt: controllerSt,
 		modelSt:      modelSt,
 		clock:        clock.WallClock,
+		provider:     provider,
 	}
 }
 
+func (s *ModelService) GetEnvironVersion(ctx context.Context) (int, error) {
```

Comment:
Would you like to have a stab at writing some documentation for this func? I find it easy to always talk about the pre and post conditions of the functions and any known failure cases that can and should be understood.

Reply:
Sure thanks for the suggestion :)